starknet_py/net/account/account_client.py

```diff
@@ -576,6 +580,65 @@ async def get_contract_nonce(
             contract_address, block_hash, block_number
         )
 
+    def sign_message(self, typed_data: TypedData) -> List[int]:
+        """
+        Sign an TypedData object for off-chain usage with the starknet private key and return the signature
+        This adds a message prefix, so it can't be interchanged with transactions
+
+        :param typed_data: TypedData object to be signed
+        :return: The signature of the TypedData object
+        """
+        return self.signer.sign_message(typed_data, self.address)
+
+    def hash_message(self, typed_data: TypedData) -> int:
+        """
+        Hash a TypedData object with pedersen hash and return the hash
+        This adds a message prefix, so it can't be interchanged with transactions
+
+        :param typed_data: TypedData object to be hashed
+        :return: the hash of the TypedData object
+        """
+        typed_data = TypedDataDataclass.from_dict(typed_data)
+        return typed_data.message_hash(self.address)
+
+    async def verify_message(self, typed_data: TypedData, signature: List[int]) -> bool:
+        """
+        Verify a signature of a TypedData object
+
+        :param typed_data: TypedData object to be verified
+        :param signature: signature of the TypedData object
+        :return: true if the signature is valid, false otherwise
+        """
+        msg_hash = self.hash_message(typed_data)
+        return await self._verify_message_hash(msg_hash, signature)
+
+    async def _verify_message_hash(self, msg_hash: int, signature: List[int]) -> bool:
+        """
+        Verify a signature of a given hash
+
+        :param msg_hash: hash to be verified
+        :param signature: signature of the hash
+        :return: true if the signature is valid, false otherwise
+        """
+        calldata = [msg_hash, len(signature), *signature]
+
+        invoke_tx = InvokeFunction(
+            contract_address=self.address,
+            entry_point_selector=get_selector_from_name("is_valid_signature"),
+            calldata=calldata,
+            signature=[],
+            max_fee=0,
+            version=0,
+            nonce=None,
+        )
+        try:
+            await self.call_contract(invoke_tx=invoke_tx)
+            return True
+        except ClientError as ex:
+            if re.search(r"Signature\s.+,\sis\sinvalid", ex.message):
+                return False
+            raise ex
```

Comment:
I'm not sure if we should just be re-raising whatever exception we got here. Perhaps `raise Exception("some message") from ex` would be a more ergonomic approach for the end user? 

cc @Solpatium

Reply:
Alright, we can leave it as it is then